repos/system_upgrade/el7toel8/actors/cupsmigrate/actor.py

```diff
@@ -0,0 +1,339 @@
+  
+      Returns boolean
+      """
+
+      try:
+        with open(include_files[0], 'r') as cupsd_conf:
+          cupsd_contents = cupsd_conf.read()
+      except IOError:
+        return False
+        
+      # concatenation of files
+      for conf in include_files[1:]:
+        if os.path.exists(conf) is True:
+          try:
+            with open(conf, 'r') as opened_conf:
+              includefile_contents = opened_conf.read()
+          except IOError:
+            return False
+          try:
+            with open(include_files[0], 'w') as cupsd_conf:
+              cupsd_contents = cupsd_contents + includefile_contents
+              cupsd_conf.write(cupsd_contents)
+          except IOError:
+            return False
+        else:
+          return False
+
+      # remove Include directives from cupsd.conf
+      found_values = self.find_values('Include', cupsd_contents)
+      for value in found_values:
+        cupsd_contents = self.remove_string('Include', value, cupsd_contents)
+
+      try:
+        with open(include_files[0], 'w') as cupsd_conf:
+          cupsd_conf.write(cupsd_contents)
+        return True
+      except IOError:
+        return False
+
+    def move(self, directives, source, target):
+      """
+      move()
+  
+      parameters:
+      - directives - list of strings
+      - source - string
+      - target - string
+  
+      returns boolean
+      """
+
+      try:
+        with open(source, 'r') as opened_source:
+          source_content = opened_source.read()
+      except IOError:
+        return False
+
+      try:
+        with open(target, 'r') as opened_target:
+          target_content = opened_target.read()
+      except IOError:
+        return False
+
+      # grap directive and its value from old config and put it into new config
+      for directive in directives:
+        regexp = '(' + directive + '\s+.+\s*\n?)'
+        pattern = compile(regexp)
+        found_lines = findall(pattern, source_content)
+        for line in found_lines:
+          try:
+            with open(target, 'w') as opened_target:
+              opened_target.write(target_content + line)
+          except IOError:
+            return False
+
+          target_content = target_content + line
+          source_content = source_content.replace(line, '')
+          try:
+            with open(source, 'w') as opened_source:
+              opened_source.write(source_content)
+          except IOError:
+            return False
+
+      return True
+
+    def process(self):
+      for model in self.consume(Cupsmigrationmodel):
+        if model.migrateable is False:
+          report_generic(
+                          title='CUPS will not be migrated.',
+                          summary='CUPS is not installed or its configuration is up-to-date.',
+                          severity='low'
+                        )
+          raise StopActorExecution('CUPS will not be migrated')
```

Comment:
No need to raise an exception here, we could just use `return`.

Reply:
Ok.